README.md

```diff
@@ -104,13 +108,25 @@ This means if the output directory has an active Terraform workspace, i.e. there
 
 ## Limitation
 
-Some Azure resources are modeled differently in AzureRM provider, which means there might be N:M mapping between the Azure resources and the Terraform resources.
+There are several limitations causing `aztfy` can hardly generate reproducible Terraform configurations.
+
+### N:M Model Mappings
+
+Azure resources are modeled differently in AzureRM provider.
 
-For example, the `azurerm_lb_backend_address_pool_address` is actually a property of `azurerm_lb_backend_address_pool`, whilst in the AzureRM provider, it has its own resource and a synthetic resource ID as `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/loadBalancers/loadBalancer1/backendAddressPools/backendAddressPool1/addresses/address1`.
+For example, the `azurerm_lb_backend_address_pool_address` is actually a property of `azurerm_lb_backend_address_pool` in Azure platform. Whilst in the AzureRM provider, it has its own resource and a synthetic resource ID as `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/loadBalancers/loadBalancer1/backendAddressPools/backendAddressPool1/addresses/address1`.
 
 Another popular case is that in the AzureRM provider, there are a bunch of "association" resources, e.g. the `azurerm_network_interface_security_group_association`. These "association" resources represent the association relationship between two Terraform resources (in this case they are `azurerm_network_interface` and `azurerm_network_security_group`). They also have some synthetic resource ID, e.g. `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/microsoft.network/networkInterfaces/example|/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/networkSecurityGroups/group1`.
 
-Currently, this tool only works on the assumption that there is 1:1 mapping between Azure resources and the Terraform resources.
+Currently, this tool only works on the assumption that there is 1:1 mapping between Azure resources and the Terraform resources. For those property-like Terraform resources, `aztfy` will just ignore them.
+
+### AzureRM Provider Validation
+
+When generating the Terraform configuration, not all properties of the resource are exported for different reasons.
+
+One reason is because there are flexible cross-property constraints defined in the AzureRM Terraform provider. E.g. `property_a` conflits with `property_b`. This might due to the nature of the API, or might be due to some deprecation process of the provider (e.g. `property_a` is deprecated in favor of `property_b`, but kept for backwards compatibility). These constraints require some properties must be absent in the Terraform configuration, otherwise, the configuration is not a valid and will fail during `terraform validate`.
```

Comment:
The paragraph of the Reason One may introduce ambiguous meanings: 1) no properties would be generated at all 2) only one of those conflicted properties would be generated. If it were the 2nd case a further ask would be raised: which one would be generated and why choose that one. Given assume the fact is the 1st case, suggest tune the description a bit.

Reply:
The way how we handle this case might keep changing. Here my intent is only to tell the reason, but not provide a commit about the implementation details. Instead, we only commit that our goal is to pass `terraform plan`.